pre-shutdown-hook

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```

Comment:
I think more elegant approach here would be adding command to the `shutdown-deferrer` binary. Like `shutdown-deferrer hook pre-shutdown` or something. Then we would have that code written in Go.

Reply:
This doesn't sound right. Should we maybe have a TODO in AZ story to rewrite it (aka. pay technical debt)?

Reply:
Sorry, not enough sleep. Yes then this is legit.

Reply:
I actually thought about upgrades story. But I didn't know this is a PM.